### ISSUE TITLE
remove / splice has bug when count is GE queue.length

### ISSUE THREAD

**Issue by txchen** [invertase/denque#9]
```
var dq = new Denque()
dq.push(1)
dq.push(2)
dq.push(3)
dq.splice(0, 3) -> return undefined
```

I think it should have same behavior like Array.splice()

**Comment by txchen** [comment]
Maybe the bug is here? 
https://github.com/invertase/denque/blob/fbfa391c733cf66cb52505969716b178825a1bd3/index.js#L226

**Comment by txchen** [comment]
@noevents FYI.

**Comment by noevents** [comment]
@txchen Thanks! I've fixed it in #10.

**Comment by Salakar** [comment]
@txchen @noevents I have merged the PR and published! Thanks
